src/v/kafka/client/client.cc

```diff
@@ -330,6 +330,30 @@ client::list_offsets(model::topic_partition tp) {
     });
 }
 
+namespace {
+ss::future<fetch_response> maybe_throw_exception(
+  shared_broker_t b, model::topic_partition tp, fetch_response res) {
+    if (res.data.error_code != error_code::none) {
+        return ss::make_exception_future<fetch_response>(
+          broker_error(b->id(), res.data.error_code));
```

Comment:
I can't convince myself that broker_error is always the right choice, but this is still an improvement.